packages/web-frontend/src/containers/MapOverlayBar/MapOverlayBar.js

```diff
@@ -208,23 +209,44 @@ const mapStateToProps = state => {
   };
 };
 
-const mapDispatchToProps = dispatch => ({
-  onExpandClick: () => dispatch(toggleMeasureExpand()),
-  onUnSelectMapOverlay: mapOverlayId => dispatch(unselectMapOverlay(mapOverlayId)),
-  onSelectMapOverlay: mapOverlayId => dispatch(selectMapOverlay(mapOverlayId)),
-  dispatch,
-});
+const mapDispatchToProps = dispatch => {
+  return {
+    onExpandClick: () => dispatch(toggleMeasureExpand()),
+    onSetMapOverlay: mapOverlayIds => dispatch(setMapOverlay(mapOverlayIds)),
+    onClearMeasure: () => dispatch(clearMeasure()),
+    dispatch,
+  };
+};
 
 const mergeProps = (stateProps, dispatchProps, ownProps) => {
-  const { dispatch } = dispatchProps;
-  const { currentMapOverlayIds } = stateProps;
+  const { dispatch, onSetMapOverlay, onClearMeasure } = dispatchProps;
+  const { currentMapOverlayIds, mapOverlayHierarchy } = stateProps;
+  const onSelectMapOverlay = mapOverlayId => {
+    const mapOverlayIds = sortMapOverlayIdsByHierarchyOrder(mapOverlayHierarchy, [
+      ...currentMapOverlayIds,
+      mapOverlayId,
+    ]);
+    onSetMapOverlay(mapOverlayIds.join(','));
```

Comment:
I think I definitely prefer props and actions to be combined in such a way here, rather than having to add new sagas! This way we can have a few well-defined actions and sagas, then client components can combine them in any way they see fit.